scripts/gitutils.py

```diff
@@ -59,19 +59,20 @@ def git_check_merge(branch1, branch2, git=None):
         git.checkout(branch2)
         is_behind = git.log('{0}..{1}'.format(branch2, branch1),
                             max_count=1).strip()
+        clean_merge = True
         if is_behind:
             try:
                 git.merge('--no-commit', '--no-ff', branch1).strip()
             except sh.ErrorReturnCode_1:
                 # git merge returns 1 when there's a conflict
-                return False
-            else:
-                return True
-            finally:
-                git.merge('--abort')
-                git.checkout(orig_branch)
-        else:
-            return True
+                clean_merge = False
+            git.merge('--abort')
```

Comment:
Yeah, the `git merge --no-comit --no-ff` makes sure to leave the merge open, regardless of success.  That's what [the linked stackoverflow comment](http://stackoverflow.com/questions/501407/is-there-a-git-merge-dry-run-option/501461#501461) alludes anyways.